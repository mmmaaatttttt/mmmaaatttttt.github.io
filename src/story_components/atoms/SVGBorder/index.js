import React, { useContext } from "react";
import PropTypes from "prop-types";
import COLORS from "utils/styles";
import { SVGContext } from "contexts";

function SVGBorder({
  borderStroke = COLORS.GRAY,
  borderWidth = 3
}) {
  const { width, height } = useContext(SVGContext);
  return (
    <rect
      x={0}
      y={0}
      width={width}
      height={height}
      stroke={borderStroke}
      strokeWidth={borderWidth}
      fill="none"
    />
  );
}

SVGBorder.propTypes = {
  borderStroke: PropTypes.string,
  borderWidth: PropTypes.number
};

export default SVGBorder;
