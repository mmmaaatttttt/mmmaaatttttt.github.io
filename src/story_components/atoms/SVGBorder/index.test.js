import React from "react";
import { render } from "@testing-library/react";
import SVGBorder from "./";

it("renders without crashing", () => {
  render(<SVGBorder />);
});

it("matches snapshot with props and children", () => {
  const { asFragment } = render(
    <SVGBorder width={200} height={200} borderWidth={6} borderStroke="purple" />
  );
  expect(asFragment()).toMatchSnapshot();
});
