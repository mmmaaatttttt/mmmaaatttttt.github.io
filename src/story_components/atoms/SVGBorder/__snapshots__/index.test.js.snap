// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props and children 1`] = `
<DocumentFragment>
  <rect
    fill="none"
    height="200"
    stroke="purple"
    stroke-width="6"
    width="200"
    x="0"
    y="0"
  />
</DocumentFragment>
`;
